### PR TITLE
rbd: correct parameter of namespace and verify it before set_namespace

### PR DESCRIPTION
**Pull request by Songweibin**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/tools/rbd/Utils.cc [hunk 1]
```diff
@@ -665,6 +665,21 @@ int init_io_ctx(librados::Rados &rados, const std::string &pool_name,
     return r;
   }
 
+  if (!namespace_name.empty()) {
+    librbd::RBD rbd;
+    std::vector<std::string> names;
+    r = rbd.namespace_list(*io_ctx, &names);
```
Comment (dillaman) [inline src/tools/rbd/Utils.cc]: For security purposes, the ```namespace_list``` cls method might be restricted to a user. Therefore, I think we would actually need a new librbd API method to assert that the namespace exists (which would invoke ```librbd::cls_client::dir_state_assert```)

**Comment by Songweibin** [comment]
@dillaman  Add a new librbd API -- `namespace_exists()`. Please take a look, thx.

File: src/librbd/api/Namespace.cc [hunk 2]
```diff
@@ -155,6 +155,34 @@ int Namespace<I>::list(IoCtx& io_ctx, vector<string> *names)
...
+    return -EINVAL;
+  }
+
+  librados::IoCtx ns_ctx;
+  ns_ctx.dup(io_ctx);
+  ns_ctx.set_namespace(name);
+
+  int r = librbd::cls_client::dir_state_assert(&ns_ctx, RBD_DIRECTORY,
+                                               cls::rbd::DIRECTORY_STATE_READY);
+  if (r == 0) {
```
Comment (Songweibin) [inline src/librbd/api/Namespace.cc]: @dillaman  Probably, it should be `r >= 0`, right?

Comment (dillaman) [reply src/librbd/api/Namespace.cc]: Eh -- that will only ever return zero on success.

Comment (Songweibin) [reply src/librbd/api/Namespace.cc]: Got it, thank you.

**Pull request merged**